### PR TITLE
Cache and validate HIBP prefix responses

### PR DESCRIPTION
**Pull request by dknauss**
### Description of the Change

Improves how the Have I Been Pwned check handles its range response: caches the k-anonymity **prefix** response rather than a per-password result, sets an explicit request timeout, and validates that the response actually arrived intact before trusting or caching it.

Previously the result was cached per full password hash, which avoids repeat checks for the same password but does not reuse the HIBP range response for different passwords sharing the same prefix. This change caches the prefix response in object cache and a transient, then checks the local suffix list for the submitted password hash.

The second commit closes the gap that caching opens. Any non-empty 200 was parsed and cached for four hours, so a response cut short at the transport boundary — or an error page served with a `200` by a captive portal or proxy — read as "not breached", and then kept reading that way for the rest of the cache lifetime. The missing tail of a truncated range is indistinguishable from a clean verdict, which is the failure worth closing before a cached body can extend it.

Now:

- **`Add-Padding` is requested**, so the response size cannot reveal how many real matches the prefix held. Padded rows carry a count of `0`, so the suffix scan also requires `count > 0` — without that check, enabling padding would produce false positives.
- **`limit_response_size`** bounds what a hijacked or proxied endpoint can stream into memory. Default 128KB (a padded range is roughly 30–50KB), filterable via `tenup_experience_hibp_max_response_bytes`, floored at 1KB so a filter cannot make every response look truncated.
- **The size check runs before `trim()`.** A capped body can end exactly on a CRLF, and trimming first would hide the truncation and make a partial range look complete enough to validate and cache.
- **The body must match the range format** — 35 hex characters, a colon, a count, one row per line — before it is used at all.
- **Only a validated body is cached.**
- Suffix comparison is case-insensitive, and `$transient_key` is hoisted so it is no longer assigned in one branch and read from another.

The HTTP request also has a short default timeout, filterable with `tenup_experience_hibp_request_timeout`.

Fails open at every step, exactly as before: an unreachable, truncated, or malformed response allows the password rather than blocking password changes while a third-party API is having a bad day.

### Benefits

- Reduces repeated outbound HIBP requests for the same SHA-1 prefix.
- Avoids caching full-hash pass/fail results.
- A truncated or malformed response can no longer be read as "not breached", or cached as such.
- Response size no longer leaks how many suffixes share a prefix.
- Prevents password-validation requests from waiting on the WordPress HTTP API default timeout.
- Keeps the existing k-anonymity request model.

### Possible Drawbacks

The prefix response may be reused for up to four hours, so newly-added HIBP entries for an already-cached prefix may not be seen until the cache expires.

Requesting padding makes responses larger — roughly 30–50KB rather than ~20KB — in exchange for a constant-size signal on the wire. The 128KB cap is well clear of that, and is filterable for anyone who wants it tighter.

### Verification Process

- `php -l includes/classes/Authentication/Passwords.php`
- `composer lint` — clean; the four warnings the full run reports are pre-existing on `develop` in `AdminCustomizations/EnvironmentIndicator.php` and are untouched here.
- `git diff --check`
- The parsing decisions were exercised against hand-built bodies: a body at and over the cap, a capped body ending exactly on CRLF, an HTML error page with a `200`, a partial trailing line, a padded (`:0`) row, a real match, and lower-case hex.

### Changelog Entry

> Fixed - Cache Have I Been Pwned prefix responses, add a request timeout, and reject truncated or malformed range responses instead of treating them as a clean result.

### Checklist:

- [x] My code follows the code style of this project.
- [x] All new and existing tests pass. _(No automated test suite exists in this repo; CI runs lint only.)_

### A note on tests

As mentioned before: this change has unit-testable logic, but the plugin ships no test harness (no PHPUnit or e2e setup; CI runs lint only), so there is nothing here to add coverage to.

That offer is now concrete. I have a dependency-free test file covering exactly the branches added here — truncation at and over the cap, the CRLF-at-boundary case, HTML bodies, partial trailing lines, padded rows, case-insensitive suffix matching, and the opt-out path. It stubs the handful of WordPress functions involved and runs as `php tests/hibp.php`, so it needs no PHPUnit and no bootstrap. Happy to open it as a separate PR if a minimal harness of that shape is welcome, or to rewrite it for PHPUnit if you would rather start there.

_AI assistance was used in drafting and reviewing this change; final authorship and verification are mine._

### CODE REVIEW

File: includes/classes/Authentication/Passwords.php [hunk 1]
```diff
@@ -424,42 +424,97 @@ protected function is_password_secure( $password ): bool {
...
+				wp_cache_set( $cache_key, $body, self::HIBP_CACHE_KEY, 4 * HOUR_IN_SECONDS );
+			}
 		}
 
-		$response = wp_remote_get( self::HIBP_API_URL . $prefix, [ 'user-agent' => '10up Experience WordPress Plugin' ] );
+		if ( ! $cached ) {
+			$response = wp_remote_get(
+				self::HIBP_API_URL . $prefix,
+				[
+					'timeout'             => (int) apply_filters( 'tenup_experience_hibp_request_timeout', 2 ),
```
[the remark this answers is not in the document]
Comment (dknauss) [reply includes/classes/Authentication/Passwords.php]: Addressed in bb14a7e — the timeout is cast to float (preserving fractional values) and falls back to the 2s default when the filtered value is not positive, so a 0/invalid return cannot disable the timeout.

**Comment by dknauss** [comment]
@darylldoyle @jamesmorrison This PR is ready for review; it caches HIBP prefix responses, requests padded results, and rejects truncated or malformed responses before using or caching them, while retaining the existing fail-open behavior. 

Copilot’s timeout feedback was addressed in bb14a7e, and the thread is resolved. 

Since this adds parsing and caching behavior in the authentication path, I can add a dependency-free test file if you'd like.
